src/policy/space.rs

```diff
@@ -199,38 +202,44 @@ impl<'a> SFTMap<'a> {
     }
 
     fn log_update(&self, space: &(dyn SFT + Sync + 'static), start: Address, bytes: usize) {
-        debug!(
-            "Update SFT for [{}, {}) as {}",
-            start,
-            start + bytes,
-            space.name()
-        );
+        debug!("Update SFT for Chunk {} as {}", start, space.name(),);
         let first = start.chunk_index();
-        let last = conversions::chunk_align_up(start + bytes).chunk_index();
         let start_chunk = chunk_index_to_address(first);
-        let end_chunk = chunk_index_to_address(last);
         debug!(
-            "Update SFT for {} bytes of [{} #{}, {} #{})",
-            bytes, start_chunk, first, end_chunk, last
+            "Update SFT for {} bytes of Chunk {} #{}",
+            bytes, start_chunk, first
         );
     }
 
     fn trace_sft_map(&self) {
         // For large heaps, it takes long to iterate each chunk. So check log level first.
         if log::log_enabled!(log::Level::Trace) {
-            // print the entire SFT map
-            const SPACE_PER_LINE: usize = 10;
-            for i in (0..self.sft.len()).step_by(SPACE_PER_LINE) {
-                let max = if i + SPACE_PER_LINE > self.sft.len() {
-                    self.sft.len()
-                } else {
-                    i + SPACE_PER_LINE
-                };
-                let chunks: Vec<usize> = (i..max).collect();
-                let space_names: Vec<&str> = chunks.iter().map(|&x| self.sft[x].name()).collect();
-                trace!("Chunk {}: {}", i, space_names.join(","));
-            }
+            trace!("{}", self.print_sft_map());
+        }
```

Comment:
The `trace!` macro itself tests if the log is enabled at the `Trace` level, therefore the `if` can be omitted.

If `print_sft_map` is only used here, and the sole purpose is to log the whole string in one atomic step, you can do this:

```rust
trace!("{}", {
    let mut res = String::new();
    for ...... {
        res.push(...);
    }
    res
});
```

Reply:
Yeah, you are right. I will remove the `if` statement. 

The reason why I put `print_sft_map()` to a separate function is that I used the function in a panic hook to print the SFT map upon panic in my debugging. I think it would be useful for debugging if we can print the SFT map in other places.